src/main/config/config.c

```diff
@@ -555,6 +567,32 @@ static void validateAndFixConfig(void)
 #if defined(TARGET_VALIDATECONFIG)
     targetValidateConfiguration();
 #endif
+
+    switch (currentControlRateProfile->rates_type) {
```

Comment:
We probably want to do this for all rates profiles, and not just for the one active at boot time. See https://github.com/betaflight/betaflight/pull/9574/files#diff-69b965793022edcd0f8430e40a617b49L225 for how this is done for PID profiles.

Reply:
Done